### PR TITLE
coot/p2p fixes

### PR DESCRIPTION
**Pull request by coot**
- connection-manager: update state of a connection when pruning
- connection-manager: fixed Show instance of TransitionTrace
- inbound-governor: removed a stale comment
- network-spec: added a missing colon
- server-tests: disable prop_never_above_hardlimit

### CODE REVIEW

File: ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs [hunk 1]
```diff
@@ -121,8 +121,10 @@ tests =
                  prop_connection_manager_pruning
   , testProperty "inbound_governor_pruning"
                  prop_inbound_governor_pruning
-  , testProperty "never_above_hardlimit"
-                 prop_never_above_hardlimit
+  {--
```
Comment (bolt12) [inline ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs:124]: Might be worth putting a TODO or a reason?

**Review by bolt12 — APPROVED** [review]
Looks good to me! :)

File: ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs (hunk 1 above)
Comment (coot) [reply ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs]: Done, linked to the issue #3487.

**Comment by coot** [comment]
hydra is down, hence some CI failures.

**Comment by coot** [comment]
Squashed into `p2p-master`

**Pull request closed without merging**